joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1064,7 +1064,8 @@ void JointTrajectoryController::fill_partial_goal(
         // Assume hold position with 0 velocity and acceleration for missing joints
         if (!it.positions.empty())
         {
-          if (has_position_command_interface_)
+          if (has_position_command_interface_ &&
+              !std::isnan(joint_command_interface_[0][index].get().get_value()))
           {
             // copy last command if cmd interface exists
             it.positions.push_back(joint_command_interface_[0][index].get().get_value());
```

Comment:
Check 2 lines below this:

`else if (has_position_state_interface_)`

Should that be an `else if` or just an `if`? Seems like it shouldn't be dependent on L1067 at all.

Reply:
Oh, I get it. L1073 is for the case where a joint is included but it doesn't have a position command interface.

Reply:
I think it should still be an `else if` because I think the idea is that if there is a command interface for said joint and it's not a NAN, then assign that joint value to be the last commanded state. Otherwise if that's not the case then just use whatever the current value is as reported from the state interface. If it's an `if`, then if the joint value was already set via the command interface, it would get overwritten

Reply:
lol, just refreshed after commenting and saw your second message. yep, you got it